backend/src/Edna/Web/Types.hs

```diff
@@ -17,4 +43,180 @@ data ExperimentalMeasurement = ExperimentalMeasurement
   , emOutlier :: Bool
   } deriving stock (Generic, Show, Eq)
 
-deriveJSON ednaAesonWebOptions ''ExperimentalMeasurement
+----------------
+-- General types
+----------------
+
+-- | A simple wrapper over 'Word'. At the data layer, we identify all entities
+-- with numeric IDs and this data type corresponds to such an ID.
+-- It has a phantom parameter type which the type of the object
+-- identified by this ID.
+newtype SqlId t = SqlId
```

Comment:
We will probably have some functions that take IDs as arguments and this way it will be easier to understand whose ID we are expecting. For example, if a function takes more than one ID argument, this phantom parameter will be very useful.

That said, currently this type resides at `Web` level and thus is not supposed to be used by business logic layer (which is below `Web`). So maybe we will move it somewhere below and will use actively. If we don't do it and keep it as `Web`-only type, maybe this phantom parameter will become useless and we'll delete it (which should be quite easy I suppose, just deletion).

These types are somewhat experimental and I imagine that something may change on Haskell side, my main goal was to keep API more or less stable from the frontend side and there it won't make any difference if we keep this phantom parameter, it will still be a number.

Reply:
Also we may have types like `ProjectId`, `CompoundId`, etc. That's another feasible option, I think it will be clearer which one is better after we implement substantial part of backend.

Reply:
>easier to understand whose ID we are expecting
Just for devs? Or it can help us in writing logic?

Reply:
I mean that if we have some `f :: SqlId Project -> SqlId Target -> IO ()`, this parameter is helpful because this type is safer than `f :: SqlId -> SqlId -> IO ()` where we can easily confuse 2 IDs (pass in wrong order).

Reply:
Make sense, but I still do not like this naming, but for now it is ok)